hyperdash/experiment.py

```diff
@@ -161,3 +165,65 @@ def end(self):
     """
     def log(self, string):
         self._logger.info(string)
+
+
+class Callbacks:
+    """Callbacks is a container class for 3rd-party library callbacks.
+    
+    An instance of Experiment is injected so that the callbacks can emit
+    metrics/logs/parameters on behalf of an experiment.
+    """
+    def __init__(self, exp):
+        self._exp = exp
+        self._callbacks = {}
+
+    @property
+    def keras(self):
+        """
+        Returns an object that implements the Keras Callback interface.
+
+        This method initializes the Keras callback lazily to to prevent
+        any possible import issues from affecting users who don't use it,
+        as well as prevent it from importing Keras/tensorflow and all of
+        their accompanying baggage unnecessarily in the case that they
+        happened to be installed, but the user is not using them.
+        """
+        cb = self._callbacks.get(KERAS)
+        # Keras is not importable
+        if cb == False:
+            return None
+        # If this is the first time, try and import Keras
+        if not cb:
+            # Check if Keras is installed and fallback gracefully
+            try:
+                from keras.callbacks import Callback as KerasCallback
+                class _KerasCallback(KerasCallback):
+                    """_KerasCallback implement KerasCallback using an injected Experiment.
+                    
+                    # TODO: Decide if we want to handle the additional callbacks:
+                    # 1) on_epoch_begin
+                    # 2) on_batch_begin
+                    # 3) on_batch_end
+                    # 4) on_train_begin
+                    # 5) on_train_end
```

Comment:
I think `on_train_end` will be key for knowing when the run ends, true?

Reply:
nah we know that already cause the user's program exits

Reply:
(or they call exp.end()